src/Integration/LocalServices/UnboundProjectFinder.cs

```diff
@@ -37,13 +40,15 @@ internal class UnboundProjectFinder : IUnboundProjectFinder
         private readonly IConfigurationProviderService configProvider;
         private readonly IProjectSystemHelper projectSystem;
         private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
 
         public UnboundProjectFinder(IServiceProvider serviceProvider, ILogger logger)
-            : this(serviceProvider, logger, new ProjectBinderFactory(serviceProvider, logger))
+            : this(serviceProvider, logger, new ProjectBinderFactory(serviceProvider, logger), new ThreadHandling())
```

Comment:
Basically, the fix is to change this class so the checking is not all done on the UI thread.

* the processing is expected to start on a background thread.
* the class switches to the UI thread each time it checks a project
i.e. there are a lot of switches back and forth. However, we`re not monopolising the UI thread, so the gold bar doesn't appear.